OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/InAppMessagesManager.kt

```diff
@@ -394,10 +395,11 @@ internal class InAppMessagesManager(
 
         // fire the external callback
         if (!_lifecycleCallback.hasSubscribers) {
-            Logging.verbose("InAppMessagesManager.messageWasDismissed: inAppMessageLifecycleHandler is null")
+            _lifecycleCallback.fireOnMain { it.onDidDismiss(InAppMessageLifecycleEvent(message)) }
```

Comment:
This seems backwards we are firing the event if we don't have subscribers. We should be firing if we do have subscribers and we shouldn't be returning in the else case